From f6d9f53369200ba277a48de3a24c377799765559 Mon Sep 17 00:00:00 2001
From: Dan Williams <dcbw@redhat.com>
Date: Wed, 3 Oct 2012 18:35:07 -0500
Subject: [PATCH] core: stop probing if a Modem has already been created for a
 port (bgo #685398)

If a port finishes probing from the first plugin, and then starts
being probed by a second plugin, and then a different port finishes
probing and creates a Modem object for that device, always let the
Modem object's plugin grab the port and ignore any other plugin.
Only one plugin may control modem ports.
---
 src/mm-manager.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/mm-manager.c b/src/mm-manager.c
index 728a115..e13d1f7 100644
--- a/src/mm-manager.c
+++ b/src/mm-manager.c
@@ -703,6 +703,14 @@ supports_callback (MMPlugin *plugin,
              * supports it.
              */
             next_plugin = existing_plugin;
+        } else if (plugin != existing_plugin) {
+            /* A modem object may have been created by one of this device's other
+             * ports while this port was getting probed; now that we have a
+             * modem, just use the modem's plugin and ignore this plugin
+             * completely.  All ports of a modem must be handled by the same
+             * plugin.
+             */
+            next_plugin = NULL;
         } else {
             mm_dbg ("(%s/%s): plugin %p (%s) existing %p (%s) info->best %p (%s)",
                     info->subsys, info->name,
-- 
1.8.1.2

